### ISSUE TITLE
`ReferenceError: document is not defined` in `Portal.svelte` in Sapper application

### ISSUE THREAD

**Issue by grantyap** [bestguy/sveltestrap#286]
Using the Offcanvas component throws `ReferenceError: document is not defined` in my Sapper application.

Here is the log:

```
ReferenceError: document is not defined
    at /Users/.../Documents/Repositories/CalPal/node_modules/sveltestrap/src/Portal.svelte:13:12
    at run (/Users/.../Documents/Repositories/CalPal/node_modules/svelte/internal/index.mjs:18:13)
    at Array.forEach (<anonymous>)
    at run_all (/Users/.../Documents/Repositories/CalPal/node_modules/svelte/internal/index.mjs:24:15)
    at Object.render (/Users/.../Documents/Repositories/CalPal/node_modules/svelte/internal/index.mjs:1386:19)
    at /Users/.../Documents/Repositories/CalPal/src/node_modules/@sapper/server.mjs:4687:54
    at Generator.next (<anonymous>)
    at fulfilled (/Users/.../Documents/Repositories/CalPal/src/node_modules/@sapper/server.mjs:128:61)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
```

This seems to be the offending code block in `Portal.svelte`:

```js
onDestroy(() => {
    document.body.removeChild(portal);
  });
```

I wrapped the statement in a check to see if the code was running on the client side:

```js
onDestroy(() => {
    if (process.browser) {
      document.body.removeChild(portal);
    }
  });
```

And it worked perfectly. The page renders without issue.

I faced a similar issue with the Dropdown component with Sveltestrap 5.0.0, I think. The nature of the problem was the same where it seemed like Sapper was trying to run `onDestroy()` code on the server side when it shouldn't be. Fortunately, this issue with the Dropdown component has since been fixed.

Looking at `Dropdown.svelte` now, it has actually implemented the same kind of fix as above:

```js
onDestroy(() => {
    if (typeof document !== 'undefined') { // Fixes `document is not defined`.
      ['click', 'touchstart', 'keyup'].forEach((event) =>
        document.removeEventListener(event, handleDocumentClick, true)
      );
    }
  });
```

In fact, using the same line in `Portal.svelte` works just as well:

```js
onDestroy(() => {
    if (typeof document !== 'undefined') {
      document.body.removeChild(portal);
    }
  });
```

I think this fix should be easy to implement, and would be really useful for any downstream projects using it.

**Comment by bestguy** [comment]
Great thank you for the find.  (I wish there was a simple way to unit test/ci the SSR issues to catch this in development, vs trying out separately in Sapper)

**Comment by bestguy** [comment]
Actually am still seeing issues, will reopen and look closer

**Comment by bestguy** [comment]
Should be fixed by #288 and released in sveltestrap@5.0.4

Thanks @grantyap for the issue and help with fix
